### ISSUE TITLE
hostedDomain causes my implementation to break

### ISSUE THREAD

**Issue by stevevance** [thephpleague/oauth2-google#55]
````
PHP Fatal error:  Uncaught League\OAuth2\Client\Exception\HostedDomainException: User is not part of domain 'http://localhost'' in /app/vendor/league/oauth2-google/src/Exception/HostedDomainException.php:13
Stack trace: [snipped]
````

I was using 2.0.1 and I upgraded to 2.2.0 and my implementing of signing in via Google broke. The error message is above. 

`http://localhost` is indeed registered with the Google Developers Console under the Credentials for this App. 

I rolled back to 2.0.1 and I upgraded to 2.1.0 and my implementation still works.

**Comment by shadowhand** [comment]
What does your `hostedDomain` configuration look like? Are you attempting to restrict logins to a single domain?

Refs #54

**Comment by shadowhand** [comment]
Well, the `hostedDomain` is just supposed to be the hostname, not the full domain. Secondly, I think you can use `hostedDomain => *` when `isOnline()` returns `false`.

**Comment by stevevance** [comment]
I completely misunderstood what `hostedDomain` option was for. I thought it was to tell Google to only allow requests from or send people back to that domain. I didn't realize it was for restricting users to a Google Apps for Business domain. 

I figured this out when I changed `http://localhost` to `localhost` and the sign in form appeared with `_____@localhost` (see screenshot). 
<img width="499" alt="screenshot 2018-03-21 14 17 18" src="https://user-images.githubusercontent.com/1330459/37732287-9c5ea98c-2d12-11e8-879f-3697c86ddfe0.png">

I have removed this option entirely from my app.
